scripts/functions.sh

```diff
@@ -55,3 +57,35 @@ for rc_file in openstack-release os-release lsb-release redhat-release; do
     source "/etc/${rc_file}"
   fi
 done
+
+## Functions -----------------------------------------------------------------
+
+function apt_artifacts_available {
+
+  CHECK_URL="${HOST_RCBOPS_REPO}/apt-mirror/integrated/dists/${RPC_RELEASE}-${DISTRIB_CODENAME}"
+
+  if curl --output /dev/null --silent --head --fail ${CHECK_URL}; then
+    return 0
+  else
+    return 1
+  fi
+
+}
+
+function configure_apt_sources {
+
+  # Replace the existing apt sources with the artifacted sources.
+
+  sed -i '/^deb-src /d' /etc/apt/sources.list
+  sed -i '/-backports /d' /etc/apt/sources.list
+  sed -i '/-security /d' /etc/apt/sources.list
+  sed -i '/-updates /d' /etc/apt/sources.list
+
+  # Add the RPC-O apt repo source
+  echo "deb ${HOST_RCBOPS_REPO}/apt-mirror/integrated/ ${RPC_RELEASE}-${DISTRIB_CODENAME} main" \
+    > /etc/apt/sources.list.d/rpco.list
+
+  # Install the RPC-O apt repo key
+  curl --silent --fail ${HOST_RCBOPS_REPO}/apt-mirror/rcbops-release-signing-key.asc | apt-key add -
+
```

Comment:
upstream's `bootstrap-ansible.sh` installs the following packages:

``` text
git-core curl gcc netcat python-minimal python-dev python3 python3-dev \
libssl-dev libffi-dev python-apt python3-apt python-virtualenv
```

maybe we can install these things first using the rpco repository without 
forcing a rewrite of the default sources list outside of the playbooks.
This way a deployer can still chose to disable artifacts without casting 
environment variables.

``` shell
apt-get update
DEBIAN_FRONTEND=noninteractive apt-get -t "${RPC_RELEASE}-${DISTRIB_CODENAME}" \
                                       -y install \
                                       git-core curl gcc netcat python-minimal \
                                       python-dev python3 python3-dev libssl-dev \
                                       libffi-dev python-apt python3-apt \
                                       python-virtualenv
```

Reply:
Sure, that may work for this - but then there are other actions which may happen prior to the sources being rewritten which put the deployment at risk. And now all developers have to be aware of this when reviewing patches. It makes it less definitive. As such, I don't think this is how it should be done.